crates/sui-framework/packages/sui-framework/tests/crypto/bls12381_tests.move

```diff
@@ -95,7 +107,481 @@ module sui::bls12381_tests {
+
+        let eight2 = bls12381::scalar_mul(&eight, &one);
+        assert!(group_ops::equal(&eight2, &bls12381::scalar_from_u64(8)), 0);
+
+        let six = bls12381::scalar_sub(&eight, &two);
+        assert!(group_ops::equal(&six, &bls12381::scalar_from_u64(6)), 0);
+
+        let minus_six = bls12381::scalar_sub(&two, &eight);
+        let three = bls12381::scalar_add(&minus_six, &bls12381::scalar_from_u64(9));
+        assert!(group_ops::equal(&three, &bls12381::scalar_from_u64(3)), 0);
+
+        let three = bls12381::scalar_div(&two, &six);
+        assert!(group_ops::equal(&three, &bls12381::scalar_from_u64(3)), 0);
+
+        let minus_three = bls12381::scalar_neg(&three);
+        assert!(group_ops::equal(&bls12381::scalar_add(&minus_three, &six), &bls12381::scalar_from_u64(3)), 0);
+
+        let minus_zero = bls12381::scalar_neg(&zero);
+        assert!(group_ops::equal(&minus_zero, &zero), 0);
+
+        let inv_three = bls12381::scalar_inv(&three);
+        assert!(group_ops::equal(&bls12381::scalar_mul(&six, &inv_three), &bls12381::scalar_from_u64(2)), 0);
+
+        let order_minus_one = bls12381::scalar_from_bytes(&ORDER_MINUS_ONE_BYTES);
+        let _ = bls12381::scalar_add(&order_minus_one, &order_minus_one);
+        let _ = bls12381::scalar_mul(&order_minus_one, &order_minus_one);
+    }
+
+    #[test]
+    fun test_valid_scalar_from_bytes() {
+        let eight = bls12381::scalar_from_u64(8);
+        let eight_from_bytes = bls12381::scalar_from_bytes(group_ops::bytes(&eight));
+        assert!(group_ops::equal(&eight, &eight_from_bytes), 0);
+
+        let zero = bls12381::scalar_zero();
+        let zero_from_bytes = bls12381::scalar_from_bytes(group_ops::bytes(&zero));
+        assert!(group_ops::equal(&zero, &zero_from_bytes), 0);
+    }
+
+    #[test]
+    #[expected_failure(abort_code = group_ops::EInvalidInput)]
+    fun test_invalid_scalar_order() {
+        let _ = bls12381::scalar_from_bytes(&ORDER_BYTES);
+    }
+
+    #[test]
+    #[expected_failure(abort_code = group_ops::EInvalidInput)]
+    fun test_invalid_scalar_too_short() {
+        let _ = bls12381::scalar_from_bytes(&SHORT_SCALAR_BYTES);
+    }
+
+    #[test]
+    #[expected_failure(abort_code = group_ops::EInvalidInput)]
+    fun test_invalid_scalar_too_long() {
+        let _ = bls12381::scalar_from_bytes(&LONG_SCALAR_BYTES);
+    }
+
+    #[test]
+    #[expected_failure(abort_code = group_ops::EInvalidInput)]
+    fun test_invalid_scalar_div() {
+        let a = bls12381::scalar_from_u64(0);
+        let b = bls12381::scalar_from_u64(10);
+        let _ = bls12381::scalar_div(&a, &b);
+    }
+
+    #[test]
+    #[expected_failure(abort_code = group_ops::EInvalidInput)]
+    fun test_invalid_scalar_inv() {
+        let a = bls12381::scalar_from_u64(0);
+        let _ = bls12381::scalar_inv(&a);
     }
-}
\ No newline at end of file
+
+
+    #[test]
+    fun test_g1_ops() {
+        let id = bls12381::g1_identity();
```

Comment:
Both this and the ops tests for the other groups below only test for consistency between the ops, so perhaps we should add some tests with hardcoded values from the groups.